src/main/java/duke/dukeobject/Budget.java

```diff
@@ -0,0 +1,122 @@
+package duke.dukeobject;
+
+import duke.exception.DukeException;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.HashMap;
+import java.util.Scanner;
+
+public class Budget {
```

Comment:
it might make more sense to create a new abstract class named DukeStorable or similar, and have this and DukeList both inherit the save and load functions that use abstract methods to convert their members to and from storage; possibly inherit undo/redo as well.